### PR TITLE
Initial DB Creation

### PR DESCRIPTION
**Pull request by RilThunder**
Some Error still inside

### CODE REVIEW

File: SQL.script [hunk 1]
```diff
@@ -1,4 +1,118 @@
-Create Database SocialMedia
```
Comment (arminhaghi) [inline SQL.script:1]: Why did you delete this part?

File: SQL.script [hunk 2]
```diff
@@ -1,4 +1,118 @@
...
-Go
-Use Database SocialMedia
-Go
+Create Table ACCOUNT
+(
+	/*
+	 SQL Server does not like max for PK
+	*/
+	Fname varchar(max) NOT NULL,
```
Comment (arminhaghi) [inline SQL.script]: We need to check with her to see if she is okay with 'MAX'

Comment (RilThunder) [reply SQL.script]: Right, I will email her right tonight

File: SQL.script [hunk 3]
```diff
@@ -1,4 +1,118 @@
...
+	/*
+	Same NO ACTION
+	*/
+	FOREIGN KEY (PostID) REFERENCES POST(PostID)  on update NO ACTION on delete NO ACTION,
+	FOREIGN KEY(Commenter) REFERENCES ACCOUNT(Email) on update NO ACTION on delete NO ACTION,
+);
+
+create table MESSAGE_THREAD
+(
+	ThreadID int NOT NULL,
```
Comment (arminhaghi) [inline SQL.script]: Same as post. Identity(1001,1)

File: SQL.script [hunk 4]
```diff
@@ -1,4 +1,118 @@
...
+	OwnerEmail varchar(40) NOT NULL,
+	PRIMARY KEY(ThreadID),
+	FOREIGN KEY(OwnerEmail) REFERENCES ACCOUNT(Email)  on update cascade on delete cascade
+);
+
+create table MESSAGE
+(
+	ThreadID int NOT NULL,
+	Sender varchar(40) NOT NULL,
+	[Date] date NOT NULL,
```
Comment (arminhaghi) [inline SQL.script]: Change Data and Time here to MsgDate, MsgTime or MessageDate, MessageTime.

File: SQL.script [hunk 5]
```diff
@@ -1,4 +1,118 @@
...
+);
+
+Create Table THREAD_PARTICIPANT
+(
+	ThreadID int NOT NULL,
+	UserEmail varchar(40) NOT NULL,
+	PRIMARY KEY(ThreadID,UserEmail),
+	FOREIGN KEY(ThreadID) REFERENCES MESSAGE_THREAD(ThreadID)  on update NO ACTION on delete NO ACTION,
+	FOREIGN KEY(UserEmail) REFERENCES ACCOUNT(Email) on update NO ACTION on delete NO ACTION
+);
```
Comment (arminhaghi) [inline SQL.script]: Always keep an empty line in the end of file, no matter what language you're writing in.

File: SQL.script (hunk 2 above)
Comment (arminhaghi) [reply SQL.script]: Remember to remind her that we're using SQL Server

File: SQL.script (hunk 1 above)
Comment (RilThunder) [reply SQL.script:1]: Oh I tried running that on a new DB in SQL Server Mangement Studio so I deleted it. Forgot to put in back in when copy-paste.
If I were to add those line, a separate query with those line will automatically create a DB right instead of me going through the old fashion way of creating a new DB and insert those SQL Script specific for the DB

File: SQL.script (hunk 4 above)
Comment (RilThunder) [reply SQL.script]: Got it

File: SQL.script (hunk 5 above)
Comment (RilThunder) [reply SQL.script]: Got it

File: SQL.script (hunk 1 above)
Comment (arminhaghi) [reply SQL.script:1]: You can always execute CREATE DATABASE XYZ, if XYZ exists, it will ignore it.

Comment (RilThunder) [reply SQL.script:1]: Right, I see. Too bad it won't do the same for the table.
There are times when I have some errors in the tables down below so I have to re-run the query. But the previous tables were already created so it did not allow. Then I have to DROP the Table to make the query work

Comment (arminhaghi) [reply SQL.script:1]: You can always highlight a part of query and then execute that only in SQL Management Studio

Comment (arminhaghi) [reply SQL.script:1]: Simply highlight the text, then hit F5

**Comment by RilThunder** [comment]
I was having trouble with the " Use Database SocialMedia" since it keeps complaining syntax error. After a while, I changed it to "Use SocialMedia" and it works great.

**Review by arminhaghi — APPROVED** [review]
Looks good.
Good job.
